### ISSUE TITLE
 Unet

### ISSUE THREAD

**Issue by huichen5796** [huichen5796/2022-projektarbeit-hui-chen#51]
Segnet -- https://github.com/codecat0/CV/blob/main/Semantic_Segmentation/SegNet/nets/segent.py
Unet -- https://github.com/pranjalrai-iitd/Fetal-head-segmentation-and-circumference-measurement-from-ultrasound-images/blob/master/Unet.py

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/173330398-9fdce2d7-7859-4737-982d-1f4656c4e0a6.png)
shit after training for a day, I got an artificial mental retardation.
This loss function is not suitable for this task, because the table area usually occupies a small proportion of the image area, so the computer chooses not to detect the table and obtains a high score.

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/173331330-a30491df-2d3c-4e62-9f62-816f4cbaf289.png)
![image](https://user-images.githubusercontent.com/79353997/173331368-1cdfd6f8-7305-492b-a596-68c137a49aba.png)
The model chooses to output only the full black image to obtain a high similarity to the label image.

**Comment by huichen5796** [comment]
![393de893d605046193e3f7c65340d15](https://user-images.githubusercontent.com/79353997/174675251-18d4e58f-aca8-4bad-adf0-9499fbbb15ae.png)

Now just trying to train for 10 epochs can get this result. goooooooooood

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/174779467-534eeec4-5a70-441f-b5dd-bef11c482734.png)

but after 100 epochs got shit

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/174855154-872dac62-c954-4944-bb28-789addc256fc.png)

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/174899989-c2088012-c9fc-4972-bef2-49852fe6eb74.png)

now use dice function, 5 epochs got this, better than bce loss

**Comment by huichen5796** [comment]
![fa75e8aa6324bd03145628ee56646e6](https://user-images.githubusercontent.com/79353997/176221380-b3b3d12b-8f88-4d5c-8e73-558276b103a1.png)

change the size of image to 512x512, The result looks so much better

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176225596-55cb0407-053f-45f9-b2a0-dc4ba5be9eb0.png)

30 epochs, It looks like it's getting worse

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176382715-0d3a0f54-772b-4120-af07-b2dc44162f48.png)

100 epochs, Es sieht nicht gut aus, es beginnt Kanten zu erkennen

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176397460-dd7b1a8c-d8b0-4bc6-afbe-852908075f16.png)

180, besser

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176398389-ec91f9e8-fd59-4332-b1f6-f4a55fd6d5e2.png)
Ein anderes Modell kann möglicherweise Textlücken erkennen

path -- /content/drive/MyDrive/model100_+20_highlr.pkl

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176412797-bcdbc32c-b9e0-4c5c-9467-3795e1647a94.png)

210 epochs, what are you doing, my computer!!!???

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176425919-ea6d8cb5-7c28-40ea-bbed-8f8ca564e433.png)

von ,,/content/drive/MyDrive/model100_+20_highlr.pkl" wurde 50 epochs traniert, Es sieht so aus, als würde der weiße Bereich zur Antwort hin schrumpfen.

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176428205-2ed179c2-394f-47c4-8e69-bd05ab08006b.png)

180 epoch， Der Textbereich wird heller

**Comment by huichen5796** [comment]
![image](https://user-images.githubusercontent.com/79353997/176738092-25f4633b-9f5d-449e-8bb6-3c03f4967469.png)

mit BCE+DICE Loss-Funktion ist die Ergebniss nach nur 40 epochs schon viel besser, die Unterschiede zwischen Text und Tabelle sind quasi schon bemerkt.

**Comment by huichen5796** [comment]
habe vorhin einen dummen Fehler gemacht. 😫😂Beim Loading des Datensatzes sind Maske und Bild nicht übereinstimmend. (also: load image1 mit, nicht mask1, sondern mask2 )

Nach Korrektur ist das Ergebnis wunderbar
![1048e29a5b47630f082748a9099127a](https://user-images.githubusercontent.com/79353997/177054535-32c2078f-d8a0-4db1-b772-ec50bfa3c15c.png)
